### ISSUE TITLE
chatterbox-qt timestamps with messages

### ISSUE THREAD

**Issue by dakaufma** [andres-erbsen/chatterbox#16]

**Comment by andres-erbsen** [comment]
The problem with timestamps is that no timestap we have is actually that great
- the "sent" timestamp can be forged by the sender
- a "server received" timestamp could be forged by the server
- a "received" timestamp (file mtime) is dependent on whether the recipient's computer was on

**Comment by dakaufma** [comment]
True, we don't have any perfect timestamp, but still... we should be able
to do well enough:

- for the most part the ratchet gives us the order or messages
    - all messages from the other person are ordered
    - ratchet advances fix ordering  between our messages and their messages
    - if both parties send messages at about the same time (resulting in no
advance it ratchet) then ... whatever, normal chat clients don't have a
perfect solution for ordering here anyway
    - if the other person is not cooperating and not advancing the ratchet
every time they get a chance to then we have bigger problems than
potentially incorrect timestamps. Although maybe if this seems to be
happening the UI can make some sort of note

So suppose you used sender timestamps for messages and check them against
the ratchet message ordering. Sure, they can forge timestamps, but they
can't change the ordering of messages. And you can't do better than that
without trusting the sender, the server, or {some combination of being
online when the message is sent and trusting the server not to delay
messages to fiddle with timestamps}. Of these things I think it is best to
just trust the sender not to mess with the timestamp (knowing that even if
they do they can't make any _major_ changes in an active conversation with
the ratchet being advanced)

**Comment by andres-erbsen** [comment]
Assuming we want to show a time, the sender time is the one we want. I don't consider showing times a priority, but I would be happy to merge a pull request.
